src/main.cc

```diff
@@ -184,9 +184,16 @@ main(const int argc, char* argv[]) {
         .is_err();
   } catch (const structopt::exception& e) {
     if (argc > 1) {
+      int subcommand_index = 1;
+      for (; subcommand_index < argc; ++subcommand_index)
+        if (argv[subcommand_index][0] != '-')
+          break;
+
       // try correcting typo
-      if (const auto sugg =
-              util::lev_distance::find_similar_str(argv[1], command_list)) {
+      if (const auto sugg = util::lev_distance::find_similar_str(
+              argv[subcommand_index], command_list
+          );
+          sugg && *sugg != argv[subcommand_index]) {
```

Comment:
```suggestion
          sugg.has_value() && sugg.value() != argv[subcommand_index]) {
```

I would recommend using methods of `std::optional` instead of `operator bool` & `operator*` which misleads pointers.